### PR TITLE
Fix segment-model tokenizers not returning token_type_ids by default

### LINKED ISSUE #48128: v5 tokenizers of segment-embedding models (ALBERT, XLNet, Funnel, Splinter, RemBERT, RoFormer, HerBERT) no longer return token_type_ids by default
### System Info

transformers 5.16.0.dev0 (main @ decba1d), Python 3.11.14, macOS arm64, torch 2.13.0 (CPU-only)

### Who can help?

@ArthurZucker @itazap

### Reproduction

```python
from transformers import AlbertTokenizer, BertTokenizer
albert, bert = AlbertTokenizer(), BertTokenizer()          # same via from_pretrained
print(sorted(albert("hello", "world").keys()))             # ['attention_mask', 'input_ids']  <- missing
print(sorted(bert("hello", "world").keys()))               # ['attention_mask', 'input_ids', 'token_type_ids']
print(albert("hello", "world", return_token_type_ids=True)["token_type_ids"])  # [0, 0, 0, 1, 1] <- capable, just dropped
```

In v4.57.0, `PreTrainedTokenizerBase.model_input_names` defaulted to `["input_ids", "token_type_ids", "attention_mask"]`; on current main it defaults to `["input_ids", "attention_mask"]` and per-model lists were re-declared. Seven tokenizers whose models consume segment embeddings were left without `"token_type_ids"`: **ALBERT, Funnel, HerBERT, RemBERT, RoFormer, Splinter, XLNet**. Since output keys are gated by `"token_type_ids" in self.model_input_names` (`src/transformers/tokenization_utils_tokenizers.py:758-759`), encoding a sentence pair silently omits `token_type_ids`; the models fall back to all-zero segment ids — silent accuracy degradation on pair classification / QA, no error. Reproduced end-to-end via `from_pretrained`/`AutoTokenizer` with a hub-style `tokenizer_config.json` (which never contains `model_input_names`), and `AlbertModel` outputs genuinely change when segment ids are supplied.

BERT, DeBERTa, TAPAS, LayoutLM kept the triple in v5; DistilBERT deliberately opts out (correct); Funnel's v5 tokenizer even preserves its special `cls:2` segment id in its `TemplateProcessing` — currently unreachable by default. The change is not mentioned in MIGRATION_GUIDE_V5.md, which suggests a propagation miss rather than intent.

### Expected behavior

Sentence-pair encodes on these seven tokenizers return `token_type_ids` by default, as in v4 and as `BertTokenizer` still does.

I have a fix ready (restore the per-model `model_input_names` triple, pattern of bert/deberta/tapas, + offline regression test) and will open a PR.

### PR DESCRIPTION
**Pull request by 2sumtech**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48129)

# What does this PR do?

Fixes #48128

**Problem**: the v5 tokenizer rewrite changed the `PreTrainedTokenizerBase.model_input_names` default from `["input_ids", "token_type_ids", "attention_mask"]` to `["input_ids", "attention_mask"]`, and seven tokenizers whose models consume segment embeddings were left without a per-model override: ALBERT, Funnel, HerBERT, RemBERT, RoFormer, Splinter, XLNet. Output keys are gated by `"token_type_ids" in self.model_input_names` (`src/transformers/tokenization_utils_tokenizers.py::_batch_encode_plus`), so sentence-pair encodes silently omit `token_type_ids` and the models fall back to all-zero segment ids — silent accuracy degradation with no error. All seven returned them by default in v4.57.0; the change is absent from MIGRATION_GUIDE_V5.md. Funnel's v5 tokenizer even preserves its special `cls:2` segment id in its `TemplateProcessing`, unreachable by default without this fix.

**Solution**: restore `model_input_names = ["input_ids", "token_type_ids", "attention_mask"]` on the seven affected tokenizers — the exact pattern already used by bert, deberta, and tapas in v5. BigBird (already declared its pair list in v4.57) and DistilBERT (deliberate opt-out) are intentionally untouched, keeping this strictly "restore v4 behavior".

**Tests**: new `TokenTypeIdsDefaultReturnTest` in `tests/tokenization/test_tokenization_utils.py` using offline toy vocabs (RoFormer split out under `@require_rjieba`), asserting the key is present by default, equals the explicit-request output, and contains segment 1. Fails on main for all seven; passes with the fix. `tests/tokenization/` shows no regressions (identical pre-existing hub-dependent failures on clean main in an offline env). `ruff check`/`ruff format` clean; `check_copies.py` and `check_modular_conversion.py` pass.

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md)?
- [x] Did you write any new necessary tests?

Disclosure: this bug was found and the patch prepared with assistance from an AI coding agent; I reviewed the change, which is intentionally limited to the seven affected `model_input_names` declarations plus a regression test, and take responsibility for it.

cc @ArthurZucker @itazap

### CODE REVIEW

File: tests/tokenization/test_tokenization_utils.py [hunk 1]
```diff
@@ -394,3 +395,46 @@ def test_import_protobuf_decode_error_does_not_mask_exceptions(self):
                     raise ValueError("real error")
                 except import_protobuf_decode_error():
                     pass
+
+
+@require_tokenizers
```
Comment (ArthurZucker) [inline tests/tokenization/test_tokenization_utils.py:400]: please remove this specialized test from here, it does not make sense. And the tests is exactly mentioning something: 
> as they did in v4

v4->v5 is a major release. can you check why this was changed please? instead of blindly fixing
